app/connectors/CodeListConnector.scala

```diff
@@ -42,62 +42,73 @@ object CodeItem {
 
 @ImplementedBy(classOf[FileBasedCodeListConnector])
 trait CodeListConnector {
-  def getProcedureCodes(locale: Locale): Map[String, ProcedureCode]
-  def getProcedureCodesForC21(locale: Locale): Map[String, ProcedureCode]
-  def getAdditionalProcedureCodesMap(locale: Locale): Map[String, AdditionalProcedureCode]
-  def getAdditionalProcedureCodesMapForC21(locale: Locale): Map[String, AdditionalProcedureCode]
+
+  type CodeMap[T <: CommonCode] = Map[String, ListMap[String, T]]
+
+  def getHolderOfAuthorisationCodes(locale: Locale): ListMap[String, HolderOfAuthorisationCode]
+  def getProcedureCodes(locale: Locale): ListMap[String, ProcedureCode]
+  def getProcedureCodesForC21(locale: Locale): ListMap[String, ProcedureCode]
+  def getAdditionalProcedureCodesMap(locale: Locale): ListMap[String, AdditionalProcedureCode]
+  def getAdditionalProcedureCodesMapForC21(locale: Locale): ListMap[String, AdditionalProcedureCode]
 
   val WELSH = new Locale("cy", "GB", "");
+  val supportedLanguages = Seq(ENGLISH, WELSH)
```

Comment:
Which languages an implementation of CodeListConnector supports is an implementation detail, does not belong in the trait IMO

Reply:
I don't agree. We are loading the codes according to a specific order (by using a `ListMap`), and this is why I also changed the return value from `Map` to `ListMap`, so it does not harm to let clients know which is the order. In addition we expect a `Locale` as input parameter so I do not see why we should not tell to clients what languages we support. As well as if we want to test the generic interface (the trait) and not the implementation.

Reply:
But the loading of codes according to a specific language order of is an implementation detail of the `FileBasedCodeListConnector`. Other implementations of this trait may not be so constrained and should not concern themselves with it. The trait is defining a contract of 'if I ask for these codes in a specified language, give me an ordered index of these code entities'.  I won't ask you to modify it but I disagree with this change.